TRANSITION.md

```diff
@@ -0,0 +1,363 @@
+
+```python
+from girder_worker_utils.transforms.girder_io import GirderClientResultTransform
+
+class GirderUploadToItem(GirderClientResultTransform):
+    def __init__(self, _id, **kwargs):
+        super(GirderUploadToItem, self).__init__(**kwargs)
+        self.item_id = _id
+
+    def transform(self, data):
+        self.gc.uploadFileToItem(self.item_id, data)
+        return self.item_id         
+```
+
+
+Unlike the ```Transform``` class,  classes derived from ```ResultTransform``` expect a value to be passed to the ```transform()``` method.  In the case of a task function that returns a single value, This value is the return value of the task function.  In the case of at ask function that returns a tuple of values,  this will be one of the positional return values (More on this later).  In the ```GirderUploadToItem``` transform we take the data and pass it on to the ```GirderClient.uploadFileToItem``` function. 
+
+_Note_ It is best practice to return a meaningful value from a ResultTransform.transform() function. That value will be forwarded on to the Celery results back end,  and used in subsequent tasks in the case of celery task chaining.
+
+To make sure the task calls this transform we must use a special keyword argument when launching the task.  That argument is the ```girder_results_hook```:
+
+```python
+from my_package.tasks import my_task
+from my_package.transforms import GirderUploadToItem
+
+# Some code that determins then item_id to upload the result too.
+
+my_task.delay(arg1, arg2,
+    girder_results_hook=[
+		GirderUploadToItem(item_id),
+	])
+```
+
+Note that ```girder_results_hook``` takes a sequence.  Should a task function return a tuple of arguments,  each argument will be passed to the coorisponding ResultTransform derived object in the sequence. E.g.:
+
+```python
+my_task.delay(arg1, arg2,
+    girder_results_hook=[
+		GirderUploadToItem(item_id), # results[0]
+		GirderUploadToItem(item_id), # results[1]
+		# ....                       # results[...]
+	])
+```
+
+This provides a generic mechanism to handle side effects on the results of a task function at the time the task is launched.
+
+
+### FAQ
+
+1. [What if my task has complex install dependencies?](HACKING.md#what-if-my-task-has-complex-install-dependencies)
+
+2. [How do I write a custom task from scratch (e.g. without the cookiecutter)](HACKING.md#what-if-my-task-has-complex-install-dependencies). A related question: What is the cookiecutter actually doing?
+
+3. What about Girder I/O? How do I push and pull data?
+  Please use Transforms to manage this process. The base Transform class is implemented in girder\_worker\_utils.transform.Transform. Currently there are several example implementations of transforms in girder\_worker\_utils.transforms.girder_io. We are hoping that as more projects implement common useful Transforms that these will be upstreamed into girder\_worker\_utils.
+
+## "Docker" Mode Legacy Tasks
+
+
+
+
+
```

Comment:
I think there is another section needed to deal with item_tasks conversions once available, which I believe will need type hinting to be in the installed task package in order to generate an adequate UI. Currently the type hinting in your example legacy code (e.g. inputs specified as "integer") is left without a replacement mechanism when ported over. This would be a layer which we should make clear is not required, but is explicitly to provide metadata for UI gen tools.

Reply:
@jeffbaumes I think this feedback is beyond the scope of this PR. Can you open it as an issue?